File: internal/model/news.go
package model

import "time"

type NewsEntry struct {
	NewsID      int       `json:"news_id"`
	Title       string    `json:"title"`
	BodyMD      string    `json:"body_md"`
	IsPublished bool      `json:"is_published"`
	CreatedOn   time.Time `json:"created_on,omitempty"`
	UpdatedOn   time.Time `json:"updated_on,omitempty"`
}

File: frontend/src/schema/news.ts
import { z } from "zod/v4";

export const schemaNewsEntry = z.object({
	news_id: z.number(),
	title: z.string(),
	body_md: z.string(),
	is_published: z.boolean(),
	created_on: z.date(),
	updated_on: z.date(),
});

export type NewsEntry = z.infer<typeof schemaNewsEntry>;
